### PR TITLE
fix: ensure all uploads avoid clashes and respect the extension

### PR DESCRIPTION
**Pull request by chamini2**
closes FEA-2115

### CODE REVIEW

File: projects/fal/tests/toolkit/file_test.py [hunk 1]
```diff
@@ -63,3 +65,36 @@ def test_gcp_storage_if_available():
...
+            pytest.skip(reason="GCLOUD_SA_JSON environment variable is not set")
+        repo = GoogleStorageRepository(bucket_name="fal_registry_image_results")
+
+    if repo == "r2":
+        r2_account_json = os.environ.get("R2_CREDS_JSON")
+        if r2_account_json is None:
+            pytest.skip(reason="R2_CREDS_JSON environment variable is not set")
+
+    file = File.from_bytes(
+        b"Hello GCP Storage!", repository=repo, file_name="hello.txt"
```
Comment (chamini2) [inline projects/fal/tests/toolkit/file_test.py:90]:
```suggestion
        b"print('Hello!')", repository=repo, file_name="hello.py"
```

File: projects/fal/tests/toolkit/file_test.py [hunk 2]
```diff
@@ -63,3 +65,36 @@ def test_gcp_storage_if_available():
...
+    )
+
+    host_and_path = file.url.split("?")[0]
+    last_path = host_and_path.split("/")[-1]
+    assert last_path.endswith(
+        ".txt"
```
Comment (chamini2) [inline projects/fal/tests/toolkit/file_test.py:96]:
```suggestion
        ".py"
```

**Review by mederka — COMMENTED** [review]
Was a file not committed by mistake?

**Comment by chamini2** [comment]
> Was a file not committed by mistake?

no, doing TDD and the tests now sucessfully fail 😄

File: .github/workflows/integration_tests.yaml [hunk 3]
```diff
@@ -37,7 +37,7 @@ jobs:
       - name: Install dependencies
         run: |
           pip install --upgrade pip wheel
-          pip install -e 'projects/fal[test]' ${{ matrix.deps }}
+          pip install -e 'projects/fal[test]' ${{ matrix.deps }} boto3
```
Comment (efiop) [inline .github/workflows/integration_tests.yaml:40]: `boto3` should be part of `[test]` extra

**Comment by efiop** [comment]
for the record: taking over...

**Comment by efiop** [comment]
Depends on https://github.com/fal-ai/fal-cdn/pull/2

**Review by kakkoyun — APPROVED** [review]

**Comment by efiop** [comment]
Depends on https://github.com/fal-ai/fal-cdn/pull/2

**Pull request merged**